apps/todomvc-remix/app/components/LoginButton.tsx

```diff
@@ -0,0 +1,27 @@
+type Props = {
+  clientId: string;
+  antiCsrfToken: string;
+};
+
+export const LoginButton = ({ antiCsrfToken, clientId }: Props) => (
+  <button
+    onClick={() => {
```

Comment:
Im old so have an allergic reaction to buttons with onClick 😆 this can just be a plain old <a>

Reply:
What's a better alternative?

Reply:
Sorry github rendered my `<a>` an an actual empty link 🤦 I mean this...

```jsx
export const LoginButton = ({ antiCsrfToken, clientId }: Props) => {
  const redirectUri = `${window.location.protocol}//${window.location.host}/api/auth/callback/google`;
  const oauthParams = {
    client_id: clientId,
    response_type: "code",
    scope: "openid",
    redirect_uri: redirectUri,
    state: `security_token=${antiCsrfToken}`,
  };

  console.log({ redirectUri });
  const queryString = new URLSearchParams(oauthParams).toString();
  const href = `https://accounts.google.com/o/oauth2/v2/auth?${queryString}`;

  return <a href={href}>Login</a>;
};
```